src/main/java/com/exadel/sober/services/UserService.java

```diff
@@ -0,0 +1,86 @@
+package com.exadel.sober.services;
+
+import com.exadel.sober.models.User;
+import com.exadel.sober.repositories.PromiseRepository;
+import com.exadel.sober.repositories.UserRepository;
+import com.exadel.sober.repositoriesImplemetation.AddictionRepositoryImpl;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.stereotype.Service;
+
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Objects;
+
+@Service
+public class UserService {
+
+    private UserRepository userRepository;
+
+    private PromiseService promiseService;
+
+    public UserService(UserRepository userRepository,
+                       PromiseService promiseService) {
+        this.userRepository = userRepository;
+        this.promiseService = promiseService;
+    }
+
+    public User addNewUser(User newUser) {
+        User userForSave = newUser;
+        userForSave.setPassword(get_SHA_512_SecurePassword(newUser.getPassword()));
+        userRepository.save(userForSave);
+        userForSave.setPassword(null);
+        return userForSave;
+    }
+
+    public User login(User loginUser) {
+
+        if (checkCredential(loginUser)){
+            User userForResponse = userRepository.getUserByEmail(loginUser.getEmail());
+            userForResponse.setPassword(null);
+            userForResponse.setPromises(promiseService.getPromisesForClient(userForResponse.getUserId()));
+            return userForResponse;
+        }
+        else {
+            return null;
+        }
+    }
+
+    public boolean existUserByEmail(String email) {
+        if (userRepository.getUserByEmail(email) != null) {
+           return true;
+        } else {
+            return false;
+        }
+    }
+
+    private boolean checkCredential(User loginUser) {
+        if (!existUserByEmail(loginUser.getEmail())) {
+            return false;
+        }
+        User user = userRepository.getUserByEmail(loginUser.getEmail());
```

Comment:
to much redundant work. You call twice userRepository.getUserByEmail(email) in ChecCredentials and after that you do it again in login() method. Instead this just try to find user by email and password by one call and if not found - throw an exception